### PR TITLE
Add option to skip multicluster envoy filter

### PR DESCRIPTION
**Pull request by aattuluri**
Please provide a description for what this PR is for:

`Context` https://github.com/istio/istio/issues/27300

[X] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml [hunk 1]
```diff
@@ -13,14 +13,15 @@ spec:
     istio: ingressgateway
   servers:
   - hosts:
-    - "*.global"
+    - "*.{{ .Values.global.multiCluster.globalDomainSuffix }}"
```
Comment (morvencao) [inline manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml:16]: Better trim space around the parameter `global.multiCluster.globalDomainSuffix`.

File: manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml [hunk 2]
```diff
@@ -47,10 +48,11 @@ spec:
       value:
         name: "envoy.filters.network.tcp_cluster_rewrite"
         config:
-          cluster_pattern: "\\.global$"
+          cluster_pattern: "\\.{{ .Values.global.multiCluster.globalDomainSuffix }}$"
```
Comment (morvencao) [inline manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml:51]: Same here, better to trim space around the parameter `global.multiCluster.globalDomainSuffix`.

File: manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml [hunk 3]
```diff
@@ -60,7 +62,7 @@ metadata:
 {{ $gateway.labels | toYaml | indent 4 }}
     release: {{ .Release.Name }}
 spec:
-  host: "*.global"
+  host: "*.{{ .Values.global.multiCluster.globalDomainSuffix }}"
```
Comment (morvencao) [inline manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml:65]: same here.

File: releasenotes/notes/27300.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: feature
+area: istioctl
+issue:
+  - 27300
+
+releaseNotes:
+  - |
+    **Added** Ability to configure domain suffix for multicluster installation
```
Comment (ericvn) [inline releasenotes/notes/27300.yaml:9]: c/Ability/ability

**Comment by howardjohn** [comment]
@stevenctl @nmittler I heard we may be planning to remove these anyways? Is this still desireable?

**Comment by aattuluri** [comment]
@howardjohn Can you point me at any docs for how this will be replaced in future.
Even if this will be taken out in the future, I would really appreciate this change as a patch into 1.6 and 1.7 as it would help multi-cluster configuration [automation](https://github.com/istio-ecosystem/admiral/issues/121). There are a few folks waiting for this. Thank you.

**Review by nmittler — APPROVED** [review]
LGTM

@howardjohn we'll still want to support `.global`, although it won't be the recommended multicluster installation.

**Comment by aattuluri** [comment]
@howardjohn @nmittler 
I have added patches for 1.7 and 1.6 release branches, please do the needful.
https://github.com/istio/istio/pull/27460
https://github.com/istio/istio/pull/27470

**Pull request merged**